packages/shared/src/hooks/useIncrementReadingRank.ts

```diff
@@ -16,14 +17,27 @@ export default function useIncrementReadingRank(): ReturnType {
   const { user } = useContext(AuthContext);
   const queryKeyRef = useRef<string[]>();
   const queryClient = useQueryClient();
-  const [clearQueries] = useDebounce(() => {
-    queryClient.invalidateQueries(queryKeyRef.current);
-  }, 100);
+  const [clearQueries] = useDebounce(
+    async (readToday = false): Promise<void> => {
+      const promises = [queryClient.invalidateQueries(queryKeyRef.current)];
```

Comment:
Is there any change this could result in accidentally clearing all queries (eg if it evaluates null?) (not sure just asking to be safe)

Reply:
Yep, if `queryKeyRef.current` is `[]` it will invalidate all queries, we should probably check if empty to avoid it.

Reply:
There's no change in behavior here from the existing one, so this risk was already there. Do we want to fix it?

Reply:
Looking at the code again, `queryKeyRef.current` should never be empty, as we assign it before calling the clearQueries function.

```typescript
const queryKey = getRankQueryKey(user);
queryKeyRef.current = queryKey;	      queryKeyRef.current = queryKey;
```

Reply:
Perhaps potentially if user is null?
Not sure but if it's easy to simply check for null value it would be appreciated.